### PR TITLE
fix(components): [cascader] initialHeight

### LINKED ISSUE #10516: [Component] [cascader] 在el-dialog组件中使用el-cascader多选属性，el-cascader组件的高度将为0
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.37`
- Element Plus Version: `2.2.20`
- Browser / OS: `UserAgent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36 Edg/107.0.1418.35`
- Build Tool: `Vite`

### Related Component

- `el-cascader`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtYnV0dG9uIEBjbGljaz1cIm1vZGFsID0gdHJ1ZVwiPueCueaIkTwvZWwtYnV0dG9uPlxuICA8ZWwtZGlhbG9nIHYtbW9kZWw9XCJtb2RhbFwiPlxuICAgIDxlbC1jYXNjYWRlclxuICAgICAgOm9wdGlvbnM9XCJvcHRpb25zXCJcbiAgICAgIDpwcm9wcz1cInsgbXVsdGlwbGU6IHRydWUgfVwiXG4gICAgLz5cbiAgPC9lbC1kaWFsb2c+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IGxhbmc9XCJ0c1wiIHNldHVwPlxuaW1wb3J0IHsgcmVmIH0gZnJvbSAndnVlJ1xuY29uc3QgbW9kYWwgPSByZWYoZmFsc2UpXG5jb25zdCBvcHRpb25zID0gW1xuICB7XG4gICAgdmFsdWU6ICdndWlkZScsXG4gICAgbGFiZWw6ICdHdWlkZScsXG4gICAgY2hpbGRyZW46IFtcbiAgICAgIHtcbiAgICAgICAgdmFsdWU6ICdkaXNjaXBsaW5lcycsXG4gICAgICAgIGxhYmVsOiAnRGlzY2lwbGluZXMnLFxuICAgICAgICBjaGlsZHJlbjogW1xuICAgICAgICAgIHtcbiAgICAgICAgICAgIHZhbHVlOiAnY29uc2lzdGVuY3knLFxuICAgICAgICAgICAgbGFiZWw6ICdDb25zaXN0ZW5jeScsXG4gICAgICAgICAgfSxcbiAgICAgICAgICB7XG4gICAgICAgICAgICB2YWx1ZTogJ2ZlZWRiYWNrJyxcbiAgICAgICAgICAgIGxhYmVsOiAnRmVlZGJhY2snLFxuICAgICAgICAgIH1cbiAgICAgICAgXSxcbiAgICAgIH0sXG4gICAgXSxcbiAgfSxcbl1cbjwvc2NyaXB0PlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJfbyI6e319)

### Steps to reproduce

1. 在el-dialog使用el-cascader组件的多选属性
2. 打开对话框
3. el-cascader组件的高度为0

### What is Expected?

el-cascader组件高度正常

### What is actually happening?

el-cascader组件高度为0

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by Simon-He95**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

fixed: #10516

### CODE REVIEW

**Comment by ryuhangyeong** [comment]
image

same phenomenon as before occurs.

File: packages/components/cascader/src/index.vue [hunk 1]
```diff
@@ -345,7 +345,7 @@ export default defineComponent({
   ],
 
   setup(props, { emit }) {
-    let inputInitialHeight = 0
+    let inputInitialHeight = 30
```
Comment (ryuhangyeong) [inline packages/components/cascader/src/index.vue:348]: image

same phenomenon as before occurs.

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtYnV0dG9uIEBjbGljaz1cIm1vZGFsID0gdHJ1ZVwiPueCueaIkTwvZWwtYnV0dG9uPlxuICA8ZWwtZGlhbG9nIHYtbW9kZWw9XCJtb2RhbFwiPlxuICAgIDxlbC1jYXNjYWRlclxuICAgICAgOm9wdGlvbnM9XCJvcHRpb25zXCJcbiAgICAgIDpwcm9wcz1cInsgbXVsdGlwbGU6IHRydWUgfVwiXG4gICAgLz5cbiAgPC9lbC1kaWFsb2c+XG4gIDxlbC1jYXNjYWRlclxuICAgIDpvcHRpb25zPVwib3B0aW9uc1wiXG4gICAgOnByb3BzPVwieyBtdWx0aXBsZTogdHJ1ZSB9XCJcbiAgLz5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5jb25zdCBtb2RhbCA9IHJlZihmYWxzZSlcbmNvbnN0IG9wdGlvbnMgPSBbXG4gIHtcbiAgICB2YWx1ZTogJ2d1aWRlJyxcbiAgICBsYWJlbDogJ0d1aWRlJyxcbiAgICBjaGlsZHJlbjogW1xuICAgICAge1xuICAgICAgICB2YWx1ZTogJ2Rpc2NpcGxpbmVzJyxcbiAgICAgICAgbGFiZWw6ICdEaXNjaXBsaW5lcycsXG4gICAgICAgIGNoaWxkcmVuOiBbXG4gICAgICAgICAge1xuICAgICAgICAgICAgdmFsdWU6ICdjb25zaXN0ZW5jeScsXG4gICAgICAgICAgICBsYWJlbDogJ0NvbnNpc3RlbmN5JyxcbiAgICAgICAgICB9LFxuICAgICAgICAgIHtcbiAgICAgICAgICAgIHZhbHVlOiAnZmVlZGJhY2snLFxuICAgICAgICAgICAgbGFiZWw6ICdGZWVkYmFjaycsXG4gICAgICAgICAgfVxuICAgICAgICBdLFxuICAgICAgfSxcbiAgICBdLFxuICB9LFxuXVxuPC9zY3JpcHQ+XG4iLCJQbGF5Z3JvdW5kTWFpbi52dWUiOiI8c2NyaXB0IHNldHVwPlxuaW1wb3J0IEFwcCBmcm9tICcuL0FwcC52dWUnXG5pbXBvcnQgeyBzZXR1cEVsZW1lbnRQbHVzIH0gZnJvbSAnLi9lbGVtZW50LXBsdXMuanMnXG5zZXR1cEVsZW1lbnRQbHVzKClcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxBcHAgLz5cbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7XG4gICAgXCJlbGVtZW50LXBsdXNcIjogXCJodHRwczovL3ByZXZpZXctMTA1MTctZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJ1bnN1cHBvcnRlZFwiXG4gIH1cbn0iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7XG4gICAgXCJ2dWVcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvcnVudGltZS1kb21AbGF0ZXN0L2Rpc3QvcnVudGltZS1kb20uZXNtLWJyb3dzZXIuanNcIixcbiAgICBcIkB2dWUvc2hhcmVkXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AdnVlL3NoYXJlZEBsYXRlc3QvZGlzdC9zaGFyZWQuZXNtLWJ1bmRsZXIuanNcIixcbiAgICBcImVsZW1lbnQtcGx1c1wiOiBcImh0dHBzOi8vcHJldmlldy0xMDUxNy1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmZ1bGwubWluLm1qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzL1wiOiBcInVuc3VwcG9ydGVkXCIsXG4gICAgXCJAZWxlbWVudC1wbHVzL2ljb25zLXZ1ZVwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQGVsZW1lbnQtcGx1cy9pY29ucy12dWVAMi9kaXN0L2luZGV4Lm1pbi5qc1wiXG4gIH0sXG4gIFwic2NvcGVzXCI6IHt9XG59IiwiZWxlbWVudC1wbHVzLmpzIjoiaW1wb3J0IHsgZ2V0Q3VycmVudEluc3RhbmNlIH0gZnJvbSAndnVlJ1xuaW1wb3J0IEVsZW1lbnRQbHVzIGZyb20gJ2VsZW1lbnQtcGx1cydcblxubGV0IGluc3RhbGxlZCA9IGZhbHNlXG5hd2FpdCBsb2FkU3R5bGUoKVxuXG5leHBvcnQgZnVuY3Rpb24gc2V0dXBFbGVtZW50UGx1cygpIHtcbiAgaWYgKGluc3RhbGxlZCkgcmV0dXJuXG4gIGNvbnN0IGluc3RhbmNlID0gZ2V0Q3VycmVudEluc3RhbmNlKClcbiAgaW5zdGFuY2UuYXBwQ29udGV4dC5hcHAudXNlKEVsZW1lbnRQbHVzKVxuICBpbnN0YWxsZWQgPSB0cnVlXG59XG5cbmV4cG9ydCBmdW5jdGlvbiBsb2FkU3R5bGUoKSB7XG4gIHJldHVybiBuZXcgUHJvbWlzZSgocmVzb2x2ZSwgcmVqZWN0KSA9PiB7XG4gICAgY29uc3QgbGluayA9IGRvY3VtZW50LmNyZWF0ZUVsZW1lbnQoJ2xpbmsnKVxuICAgIGxpbmsucmVsID0gJ3N0eWxlc2hlZXQnXG4gICAgbGluay5ocmVmID0gJ2h0dHBzOi8vcHJldmlldy0xMDUxNy1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmNzcydcbiAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2xvYWQnLCByZXNvbHZlKVxuICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignZXJyb3InLCByZWplY3QpXG4gICAgZG9jdW1lbnQuYm9keS5hcHBlbmQobGluaylcbiAgfSlcbn0iLCJfbyI6eyJzaG93SGlkZGVuIjp0cnVlLCJzdHlsZVNvdXJjZSI6Imh0dHBzOi8vcHJldmlldy0xMDUxNy1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmNzcyJ9fQ==)

Comment (ryuhangyeong) [reply packages/components/cascader/src/index.vue:348]: Also, the cascader component has size props, so each props (small, large) has a different height.

Comment (Simon-He95) [reply packages/components/cascader/src/index.vue]: I made some modifications

File: packages/components/cascader/src/index.vue [hunk 2]
```diff
@@ -726,9 +726,15 @@ export default defineComponent({
 
     watch(presentText, syncPresentTextValue, { immediate: true })
 
+    const getInitialHeight = () => {
+      if (props.size === 'large') return 38
+      if (props.size === 'small') return 22
+      return 30
```
Comment (init-qy) [inline packages/components/cascader/src/index.vue:732]: I think we should use css variable to replace hardcode here
#9982 #10308

Comment (Simon-He95) [reply packages/components/cascader/src/index.vue:732]: His original logic also has inputInner.offsetHeight
 `inputInitialHeight = inputInner.offsetHeight || initialHeight`
cssVar isn't it an another form hardcode
Just personal feeling, there is more burden to find dom to inject style, I don't think it seems to have any advantages over the current method

**Comment by holazz** [comment]
Hi, @Simon-He95 This bug was introduced by #10214, I created a [PR](https://github.com/element-plus/element-plus/pull/10538) to fix it.

**Comment by Simon-He95** [comment]
> Hi, @Simon-He95 This bug was introduced by #10214, I created a [PR](https://github.com/element-plus/element-plus/pull/10538) to fix it.

OK

**Pull request closed without merging**